### PR TITLE
added signal for RegistrationForm created

### PR DESCRIPTION
**Pull request by openprojects**
Add a signal sent when a RegistraionForm has been created

### CODE REVIEW

File: indico/core/signals/core.py [hunk 1]
```diff
@@ -82,6 +82,11 @@
 other way.
 """)
 
+registration_form_created = _signals.signal('registration-form-created', """
```
Comment (ThiefMaster) [inline indico/core/signals/core.py]: Should be in the same module as the other registration related signals

Comment (ThiefMaster) [inline indico/core/signals/core.py]: indico/core/signals/event/registration.py

**Comment by ThiefMaster** [comment]
Commit message should use present tense, e.g. "Add registration_form_created signal"

File: indico/core/signals/core.py (hunk 1 above)
Comment (openprojects) [inline indico/core/signals/core.py]: ok. I'moving it

**Comment by openprojects** [comment]
move signal from core to registration section

File: indico/modules/events/registration/controllers/management/regforms.py [hunk 2]
```diff
@@ -68,6 +69,7 @@ def _process_POST(self):
             create_personal_data_fields(regform)
             db.session.add(regform)
             db.session.flush()
+            signals.registration_form_created.send(regform)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/regforms.py]: Needs to be updated too

File: indico/core/signals/event/registration.py [hunk 3]
```diff
@@ -30,3 +30,8 @@
 registration_deleted = _signals.signal('registration-deleted', """
 Called when a registration is removed.  The `sender` is the registration.
 """)
+
+registration_form_created = _signals.signal('registration-form-created', """
+Called when a new registration form is created.  The `sender` is the
+`RegistrationForm` object.
+""")
```
Comment (ThiefMaster) [inline indico/core/signals/event/registration.py]: Trailing line break is missing

File: indico/modules/events/registration/controllers/management/regforms.py (hunk 2 above)
Comment (openprojects) [inline indico/modules/events/registration/controllers/management/regforms.py]: yes, I see the signal added on line 72 and 98 in my branch... am I missing something?

**Comment by mic4ael** [comment]
You can easily squash those commits into one (using git)

File: indico/modules/events/registration/controllers/management/regforms.py (hunk 2 above)
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/regforms.py]: We don't import all signals in `indico.core.signals` so you need `signals.event.registration_form_created` here.

File: indico/modules/events/registration/controllers/management/regforms.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 from __future__ import unicode_literals
 
 from flask import redirect, flash, session
+from indico.core import signals
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/regforms.py]: this should be in the next line (i.e. an empty line after the flask import) - we group imports like this:

```
__future__

imports from the stdlib

imports from thirdparty libs (like flask)

imports from indico itself
```

**Comment by ThiefMaster** [comment]
Tested it and it works fine. Can you amend your commit to remove the (unnecessary) "Fixed signal call syntax" line in the message? Once that's done I'll merge it.

**Comment by ThiefMaster** [comment]
Are you guys based on v1.9.6 or master? In case of v1.9.6 I could merge it there (the PR will then close automatically the next time we merge 1.9.6 into master) since it's a very small/simple PR that won't break anything.

**Comment by openprojects** [comment]
Master, but we are just developing on it (our "production" env is 1.9.2 
and we are migrating all our custom staff to 1.9.latest in plugins the 
"good way").
So I think it perfect to have it in 1.9.6 :)
Thank you Adrian,

Giorgio

Il 2016-01-25 15:59 Adrian ha scritto:

**Comment by openprojects** [comment]
Amend: done... but I'm not sure the result is the expected one...

**Comment by ThiefMaster** [comment]
Uh, why did you do a merge? You never want `git merge` (except with `--rebase`) in feature branches.

**Comment by ThiefMaster** [comment]
f7387f74811c980c5017989dc31772e5d7c16516 is the proper commit, simply do `git reset --hard f7387f74811c980c5017989dc31772e5d7c16516` and then `git push -f`

**Comment by openprojects** [comment]
It looks like a merge, but I honestly didn't call for it! :) 
OK... I'll follow your instructions and cross my fingers

**Comment by openprojects** [comment]
ooook. this looks better now.

**Comment by ThiefMaster** [comment]
Merged 2aacb841fb205223c032e06c85c192614e8fe334 into v1.9.6

**Pull request closed without merging**